pints/_mcmc/__init__.py

```
@@ -645,6 +680,25 @@ def run(self):
             active = list(range(self._n_chains))
             n_samples = [0] * self._n_chains
 
+        # initialisation
+        if self._x0_isfunction:
+            if self._single_chain:
+                initialised_finite, init = self._sample_x0(active, evaluator)
+            else:
+                chain_r = list(range(self._n_chains))
+                initialised_finite, init = self._sample_x0(chain_r, evaluator)
+            if not initialised_finite:
```

I'd just raise this error in `_sample_x0` instead of returning a bool to check?